components/drivetrain.py

```diff
@@ -1,21 +1,85 @@
+import enum
+
 import wpilib
+import ctre
+from common.motion_profiles import MotionProfile
+from common.pid import PIDController, PIDCoefficients
+
+__all__ = ["Drivetrain"]
+
+
+class ProfileType(enum.Enum):
+    displacement = True
+    rotation = False
 
 
 class Drivetrain:
     robot_drive = wpilib.RobotDrive
-    rotation = 0
-    forward = 0
+    gyro = wpilib.ADXRS450_Gyro
+    encoder_motor = ctre.CANTalon
+
+    def __init__(self):
+        self.rotation = 0
+        self.forward_speed = 0
+        self.motion_profile = None
+        self.pid = None
+        self.profile_input = None
 
     def forward_at(self, speed):
-        self.forward = speed
+        self.forward_speed = speed
 
     def turn_at(self, speed, squaredInputs=False):
         self.rotation = speed
         if squaredInputs:
             self.rotation = speed**2 if speed >= 0 else -(speed**2)
 
+    def forward(self, feet=0, inches=0, meters=0):
+        distance = (inches * 0.0254) + (feet * 0.304) + meters
+
+        self.profile_input = ProfileType.displacement
+        self.profile = MotionProfile.generate_motion_profile(
+            acceleration_time=1,
+            deceleration_time=1,
+            max_speed=1,
+            target_distance=distance)
+
+        coefs = PIDCoefficients(p=1.0, i=0.0, d=0.0)
+        self.pid = PIDController(coefs, 1, -1)
+
+    def rotate(self, degrees=0):
+        self.profile_input = ProfileType.rotation
+        self.profile = MotionProfile.generate_motion_profile(
+            acceleration_time=1,
+            deceleration_time=1,
+            max_speed=1,
+            target_distance=degrees)
+
+        coefs = PIDCoefficients(p=1.0, i=0.0, d=0.0)
+        self.pid = PIDController(coefs, 1, -1)
+
+    def get_encoder_position(self):
```

Comment:
What's this for?

Reply:
I think you're reviewing an outdated version. Try switching to the latest commit.